webpack/assets/javascripts/react_app/components/componentRegistry.js

```diff
@@ -40,37 +45,49 @@ const componentRegistry = {
   },
 
   getComponent(name) {
+    if (!this.registry[name]) {
+      throw new Error(
+        `Component not found: ${name} among ${this.registeredComponents()}`
+      );
+    }
+
     return this.registry[name];
   },
 
+  wrapperFactory() {
+    return new WrapperFactory();
```

Comment:
does this happen once per document load? multiple times? I'm guess I don't understand the background for all of this change...

Reply:
It's called once per component mount when it's mounted into erb or angular. I added that to enable plugins using wrappers easily. An example in Katello is here:
https://github.com/Katello/katello/pull/7001/files#diff-5bcfdae0ace18ad3702fbb46115e768dR19